src/SingleCellAnnotator.imjoy.html

```diff
@@ -107,50 +107,71 @@
       img.src = url;
   });
 };
-
-const colormap = {'Microtubule': 0, 'Protein': 1, 'ER': 0, 'Nuclei': 2, 'Mask': 3}
-
 async function composeChannels(channels, channel_settings, alpha, brightness){
-  if(alpha<0) alpha = 0;
-  if(alpha>1) alpha = 1;
-  brightness = brightness || 1.0;
-  const rgb_channels = []
-  const canvas = document.createElement('canvas') 
-  for(let k in channel_settings){
-    if(channels[k] && colormap.hasOwnProperty(k)){
-      if(channel_settings[k].show){
-        const imgData = await getImageData(channels[k]);
-        rgb_channels[colormap[k]] = imgData;
-        canvas.width = imgData.width
-        canvas.height = imgData.height
+  const color_channels = [
+    ['Microtubule', 'ER'],
+    ['Protein', 'ER'],
+    ['Nuclei'],
+    ['Mask']
+  ]
+  // replace channel name with data
+  for(let i=0;i<color_channels.length;i++){
+    for(let j=0;j<color_channels[i].length;j++){
+      const ch_name = color_channels[i][j]
+      if(channel_settings[ch_name].show){
+        color_channels[i][j] = await getImageData(channels[ch_name])
       }
     }
-    if(!colormap.hasOwnProperty(k)){
-      console.warn('No color map defined for ' + k)
+    // filter out empty channels
+    color_channels[i] = color_channels[i].filter(item=>typeof(item)!=='string')
+  }
+  // set canvas
+  const canvas = document.createElement('canvas')
+  for(let i=0;i<color_channels.length;i++){
+    if(color_channels[i].length){
+      canvas.width = color_channels[i][0].width
+      canvas.height = color_channels[i][0].height
+      break
     }
   }
   const ctx = canvas.getContext('2d');
   const imgData = ctx.getImageData(0, 0, canvas.width, canvas.height)
   const channel_offset = canvas.width * canvas.height * 4
   let i, j, v;
   const data = imgData.data
-  // set color channels
+  // get rgb_channels data
+  if(alpha<0) alpha = 0;
+  if(alpha>1) alpha = 1;
+  brightness = brightness || 1.0;
+  const rgb_channels = []
   for(j=0;j<3;j++){
-    if(rgb_channels[j]){
-      const src = rgb_channels[j].data;
-      for(i=0;i<channel_offset;i++){
-        if(brightness===1){
-          data[i*4+j] = src[i]
-        }
-        else{
-          v = src[i]*brightness;
-          data[i*4+j] = v>255?255:parseInt(v);
-        }
+    if(!color_channels[j].length){
+      continue
+    }else{
+      rgb_channels[j] = color_channels[j][0]
```

Comment:
You don’t need rgb_channels at all, the point is to only use color_channels and one for loop.